undertow/src/main/java/org/wildfly/extension/undertow/Host.java

```diff
@@ -133,26 +135,42 @@ private String getDeployedContextPath(DeploymentInfo deploymentInfo) {
     @Override
     public void start(StartContext context) throws StartException {
         suspendControllerInjectedValue.getValue().registerActivity(suspendListener);
-        if(suspendControllerInjectedValue.getValue().getState() == SuspendController.State.RUNNING) {
+        SuspendController.State state = suspendControllerInjectedValue.getValue().getState();
+        if(state == SuspendController.State.RUNNING) {
             defaultHandler.setSuspended(false);
         } else {
             defaultHandler.setSuspended(true);
         }
         ControlledProcessStateService controlledProcessStateService = controlledProcessStateServiceInjectedValue.getValue();
         //may be null for tests
         if(controlledProcessStateService != null && controlledProcessStateService.getCurrentState() == ControlledProcessState.State.STARTING) {
-            gateHandlerWrapper = new GateHandlerWrapper(queueRequestsOnStart ? -1 : defaultResponseCode);
-            controlledProcessStateService.addPropertyChangeListener(new PropertyChangeListener() {
-                @Override
-                public void propertyChange(PropertyChangeEvent evt) {
-                    controlledProcessStateService.removePropertyChangeListener(this);
-                    if(gateHandlerWrapper != null) {
-                        gateHandlerWrapper.open();
-                        gateHandlerWrapper = null;
-                    }
-                    rootHandler = null;
+            // Non-graceful is ControlledProcessState.State.STARTING && SuspendController.State.RUNNING. We know from above
+            // that we are STARTING, so we just need to check that state of the SuspendController
+            boolean nonGraceful = state == SuspendController.State.RUNNING;
+
+            int statusCode = defaultResponseCode;
+            if (nonGraceful && queueRequestsOnStart == null) {
+                UndertowLogger.ROOT_LOGGER.debug("Running in non-graceful mode and " + QUEUE_REQUESTS_ON_START.getName() +
+                        " not explicitly set. Requests will not be queued.");
+            } else {
+                if (queueRequestsOnStart == null || Boolean.TRUE == queueRequestsOnStart) {
```

Comment:
I suggest using `Boolean.TRUE.equals(queueRequestsOnStart)` to compare values, not instance references.

Reply:
Bah. Good catch. :)